rfc/rfc015-node-identity.md

```diff
@@ -0,0 +1,126 @@
+Measures to keep data private can range from encrypting data to sending data over a specific connection.
+The TLS connection and certificate exchange form the basis of the node identification.
+The *Subject Alternative Name* of the certificate is matched against the *serviceEndpoint* of a DID Document service.
+The requirement for the additional service in the DID Document also makes it possible to find other nodes and thus enable service discovery of Nuts nodes.
+
+Private data, like authorization credentials, have DIDs as issuer and subject. To enable private addressing, a DID will be encrypted with its public encryption key.
+Only the owner of the private key can decrypt this. Doing this on the individual DID level will be very inefficient.
+Each node would have to try all the private keys it manages to find out if a piece of data is addressed to it.
+To make this more efficient this RFC links individual DIDs to the service provider DID.
+An encrypted address would then only have to be decrypted with the service provider DID.
+
+### Status
+
+This document is currently a draft.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1. Introduction
+
+When all participants of a network share the same public data, there's no need to know which node services a particular organization.
+This changes when data becomes private. Private data needs to be routed to the correct node.
+In order for that to work, there's a need to identify each node and link that identity to services organizations.
+
+Two things need to be resolved:
+
+1. Node identity for peers that are connected. Network connections are the only interaction a node has to another node, so the connecting peer needs to be identified.
+2. The DID subject that needs to receive the private data needs to be linked to a DID that has a node identity.
+
+For the first point TLS and certificates will be used as basis.
+
+## 2. Terminology
+
+* **DID**: [Decentralized Identifiers](https://www.w3.org/TR/did-core/).
+* **SAN**: [subject alternative name](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6)
+* **Service provider**: organization that runs the Nuts node and manages private keys for other organizations.
+* **VC**: [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+
+Other terminology comes from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Identification
+
+### 3.1 Certificates
+
+TLS certificates are linked to a domain name via a SAN extension.
+
+```
+X509v3 extensions:
+  X509v3 Subject Alternative Name:
+    DNS:example.com
+```
+
+This domain is checked by a client that connects to a server with that domain. The certificate the server uses must match the domain the client dialed.
```

Comment:
this is TLS 101 (hostname check) and should be considered standard practice? Or is it just to inform the reader on how it works in general?

Reply:
Just to remind that there's a difference between being a client and a server.